src/Pandell.PackageRunner/Pandell.PackageRunner.csproj

```diff
@@ -0,0 +1,109 @@
+      <HintPath>..\..\nuget_modules\NuGet.Core.2.8.6\lib\net40-Client\NuGet.Core.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="System" />
+    <Reference Include="System.ComponentModel.Composition" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Web.Extensions" />
+    <Reference Include="WindowsBase" />
+  </ItemGroup>
+  <ItemGroup>
+    <Compile Include="AssemblyChecker.cs" />
+    <Compile Include="Extensions.cs" />
+    <Compile Include="FileLogging.cs" />
+    <Compile Include="NugetCredentialProvider.cs" />
+    <Compile Include="Program.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="..\AssemblyVersionInfo.cs">
+      <Link>Properties\AssemblyVersionInfo.cs</Link>
+    </Compile>
+    <Compile Include="Token.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="..\..\nuget_modules\Microsoft.Web.Xdt.2.1.1\lib\net40\Microsoft.Web.XmlTransform.dll">
+      <Link>Properties\Microsoft.Web.XmlTransform.dll</Link>
+    </EmbeddedResource>
+    <EmbeddedResource Include="..\..\nuget_modules\NuGet.Core.2.8.6\lib\net40-Client\NuGet.Core.dll">
+      <Link>Properties\NuGet.Core.dll</Link>
+    </EmbeddedResource>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <Target Name="GetPublicKeyToken" BeforeTargets="BeforeBuild">
+    <GetFrameworkSdkPath>
+      <Output TaskParameter="Path" PropertyName="WindowsSdkPath" />
+    </GetFrameworkSdkPath>
+    <Exec Command="WHERE /r &quot;$(WindowsSdkPath.TrimEnd('\\'))&quot; sn" ConsoleToMSBuild="true">
+        <Output TaskParameter="ConsoleOutput" PropertyName="SNPath" />
+    </Exec>
+    <PropertyGroup>
+      <SNPath>$([System.Text.RegularExpressions.Regex]::Replace('$(SNPath)', ';.*', ''))</SNPath>
+    </PropertyGroup>
+    <Exec Command="&quot;$(SNPath)&quot; -p &quot;$(AssemblyOriginatorKeyFile)&quot; Pandell.PackageRunner.pub" />
+    <Exec Command="&quot;$(SNPath)&quot; -q -t Pandell.PackageRunner.pub" ConsoleToMSBuild="true">
+        <Output TaskParameter="ConsoleOutput" PropertyName="TokenFile" />
+    </Exec>
+    <PropertyGroup>
+      <TokenFile>$([System.Text.RegularExpressions.Regex]::Replace('$(TokenFile)', 'Public key token is ', ''))</TokenFile>
+    </PropertyGroup>
+    <ReadLinesFromFile File="Token.cs">
+      <Output TaskParameter="Lines" PropertyName="TokenCode" />
+    </ReadLinesFromFile>
+    <PropertyGroup>
+      <TokenCode>$([System.Text.RegularExpressions.Regex]::Replace('$(TokenCode)', ';', '%0D%0A'))</TokenCode>
+      <TokenCode>$([System.Text.RegularExpressions.Regex]::Replace('$(TokenCode)', '\".*\"%0D%0A', '"$(TokenFile)";'))</TokenCode>
+    </PropertyGroup>
+    <Message Text="$(TokenCode)" />
+    <WriteLinesToFile File="Token.cs" Lines="$(TokenCode)" Overwrite="true" />
```

Comment:
Nice target. Couple of notes:

- `UpdatePublicKeyToken` would be a better name than the current `GetPublicKeyToken` IMO

- A C# MSBuild task ([example](https://github.com/pandell/PackageRunner/blob/project-restructure/tools/Pandell.targets#L683-L708)) for the update part would be more readable IMO and would make it very simple to modify the C# generated file only when token changes.